crates/rpc/src/v02/method/get_transaction_by_hash.rs

```diff
@@ -41,13 +49,36 @@ pub async fn get_transaction_by_hash(
         let db_tx = db.transaction().context("Creating database transaction")?;
 
         // Get the transaction from storage.
-        StarknetTransactionsTable::get_transaction(&db_tx, input.transaction_hash)
+        let tx = StarknetTransactionsTable::get_transaction(&db_tx, input.transaction_hash)
             .context("Reading transaction from database")?
-            .ok_or(GetTransactionByHashError::TxnHashNotFound)
-            .map(|tx| tx.into())
+            .map(Transaction::from);
+
+        anyhow::Ok(tx)
     });
 
-    jh.await.context("Database read panic or shutting down")?
+    if let Some(tx) = jh.await.context("Database read panic or shutting down")?? {
+        return Ok(tx);
+    }
+
+    use starknet_gateway_client::ClientApi;
+    let tx = context
+        .sequencer
+        .transaction(input.transaction_hash)
+        .await
+        .context("Transaction not found locally, querying gateway")?;
+
+    use starknet_gateway_types::reply::Status as GatewayStatus;
+    match tx.status {
```

Comment:
nit: "guardian pattern" to un-nest positive flow?

```suggestion
    if matches!(tx.status, GatewayStatus::NotReceived) {
        return Err(GetTransactionByHashError::TxnHashNotFound);
    }
    Ok(tx
            .transaction
            .map(Transaction::from)
            .context("Gateway transaction reply is missing body")?)
```